internal/core/metadata/application/device.go

```diff
@@ -56,12 +62,22 @@ func AddDevice(d models.Device, ctx context.Context, dic *di.Container) (id stri
 		addedDevice.Id,
 		correlation.FromContext(ctx),
 	)
-	go addDeviceCallback(ctx, dic, dtos.FromDeviceModelToDTO(d))
+
+	device := dtos.FromDeviceModelToDTO(d)
+	go addDeviceCallback(ctx, dic, device)
+
+	if err := publishDeviceSystemEvent(common.DeviceSystemEventActionAdd, d, lc, dic); err != nil {
```

Comment:
why not using async call?

Reply:
we don't handle the error and we usually use async call to publish data to message bus.
if the connection to the message bus is blocking due to network issue or something, the Device Add/Update/Delete will be blocked, too.
that's why we use `go addDeviceCallback(ctx, dic, device)` to prevent any unexpected situation.

Reply:
OK

Reply:
Added go func's